GAE/src/org/akvo/flow/xml/XmlForm.java

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright (C) 2019 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.xml;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.waterforpeople.mapping.app.gwt.client.survey.QuestionGroupDto;
+import org.waterforpeople.mapping.app.gwt.client.survey.SurveyDto;
+
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+/*
+ * Class for working with a form XML file like this:
+ * <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+ * <survey name="Brand new form four" defaultLanguageCode="en" version='4.0' app="akvoflowsandbox"
+ *  surveyGroupId="41213002" surveyGroupName="Brand new" surveyId="43993002">
+ * <questionGroup><heading>Foo</heading>
+ * ...
+ * </questionGroup>
+ * <questionGroup><heading>Bar</heading>
+ * ...
+ * </questionGroup>
+ * </survey>
+ */
+
+@JacksonXmlRootElement(localName = "survey") public final class XmlForm {
+
+    @JacksonXmlElementWrapper(localName = "questionGroup", useWrapping = false)
+    private XmlQuestionGroup[] questionGroup;
```

Comment:
Its better as an `Array` than as a `List`?

Reply:
I don't think the framework supports representing it (several identical tags) as a list.